packages/x-date-pickers-pro/src/internal/models/range.ts

```diff
@@ -0,0 +1,13 @@
+import { FieldSection } from '@mui/x-date-pickers/internals-fields';
+
+export type DateRange<TDate> = [TDate | null, TDate | null];
+export type NonEmptyDateRange<TDate> = [TDate, TDate];
+
+export interface CurrentlySelectingRangeEndProps {
```

Comment:
In the new `DateRangeCalendar` component I'm calling it `currentDatePosition`
This interface is not used on the fields or on the new pickers so it will just go away once we drop the old pickers :+1: